phoenix-core/src/main/java/org/apache/phoenix/execute/ClientScanPlan.java

```diff
@@ -106,12 +106,18 @@ public ExplainPlan getExplainPlan() throws SQLException {
     }
 
     @Override
-    public QueryPlan limit(Integer limit) {
-        if (limit == this.limit || (limit != null && limit.equals(this.limit)))
+    public QueryPlan limit(Integer limit, Integer offset) {
+        if (limit == this.limit || (limit != null && limit.equals(this.limit))) {
```

Comment:
This does definitely have ripple effects throughout, but I had done this on the basis that SQL only allows an offset if their is an imposed limit (since an offset doesn't make any logical sense without a limit).